corehq/apps/saved_reports/tasks.py

```diff
@@ -43,14 +45,45 @@ def send_delayed_report(report_id):
 def send_report(notification_id):
     notification = ReportNotification.get(notification_id)
 
+    # Intentionally let this raise an IndexError if there's no report.
+    # Shouldn't ever happen, and if it does, it means there's something wrong
+    # with our queuing or concurrency model.
+    record = ScheduledReportRecord.objects.filter(
+        state=ScheduledReportRecord.States.queued,
+        scheduled_report_id=notification_id,
+    ).order_by('-scheduled_for')[0]
+
+    # Mark all previous queued record as skipped
```

Comment:
This is new functionality right. Up till now it will still process all of those reports.

Reply:
Not exactly. Up until now (in terms of what's actually running in production), it would only check for reports scheduled to fire at the most recent 15-minute mark, so anything before then that hadn't been processed was silently skipped. Now, everything scheduled to fire since the last check is queued, but if a report was scheduled to fire more than once since the last time, all but the last are marked as skipped. (This should be exceedingly rare—it would only happen if the time since the last check was longer than a day, and we check every 15 minutes.)